src/main/java/com/jfrog/ide/common/tree/DescriptorFileTreeNode.java

```diff
@@ -0,0 +1,39 @@
+package com.jfrog.ide.common.tree;
+
+import java.util.Collection;
+
+public class DescriptorFileTreeNode extends FileTreeNode {
+    public DescriptorFileTreeNode(String filePath) {
+        super(filePath);
+    }
+
+    public void addDependency(Artifact dependency) {
+        Artifact clonedDep = (Artifact) dependency.clone();
+        add(clonedDep);
+        if (clonedDep.getTopSeverity().isHigherThan(topSeverity)) {
+            topSeverity = clonedDep.getTopSeverity();
+        }
+        sortChildren();
+    }
+
+    public void addDependencies(Collection<Artifact> dependencies) {
+        for (Artifact dependency : dependencies) {
+            add(dependency);
```

Comment:
Shouldn't need to clone the artifact here?

Reply:
That's a good question.
addDependencies is used only in SingleDescriptorScanManager, so there's one descriptor and all of the Artifacts are its children.
On the other hand, addDependency is used only in MavenScanManager, where there might be more than one descriptor, and each artifact can be a child of more than one of them, so it must be cloned.
I'll move the cloning to MavenScanManager and add comments to make it all clearer.